### ISSUE TITLE
Overloaded function's implementation is incorrectly shown as a signature

### ISSUE THREAD

**Issue by glacials** [microsoft/pylance-release#1498]
## Environment data

-   Language Server version: 2021.6.944021595
-   OS and version: Ubuntu 20.04.2 LTS on WSL2 on Windows 10
-   Python version (& distribution if applicable, e.g. Anaconda): 3.9.4 (CPython)

## Expected behaviour

If I use [`typing.overload`](https://docs.python.org/3/library/typing.html#typing.overload) to define multiple signatures for a function, there are two accepted ways to do it.

I can do it the way the `typing` module [tells me](https://docs.python.org/3/library/typing.html#typing.overload), where the implementation of the function uses no type hints:
```python
from typing import overload

@overload
def identity(a: int) -> int:
    ...

@overload
def identity(a: str) -> str:
    ...

def identity(a):
    return a
```

`mypy` also [documents a second way](https://mypy.readthedocs.io/en/stable/more_types.html#function-overloading), where the implementation uses "combined" type hints:

```python
from typing import overload, Union

@overload
def identity(a: int) -> int:
    ...

@overload
def identity(a: str) -> str:
    ...

def identity(a: Union[str, int]) -> Union[str, int]:
    return a
```

With either of these ways, I expect the IntelliSense hints for the function signatures to give me two options:

- `identity(a: int) -> int`
- `identity(a: str) -> str`

## Actual behaviour

Instead, it gives me three options.

First example:
![image](https://user-images.githubusercontent.com/438911/123344085-f6fa0580-d507-11eb-8979-ae263594a0d3.png)

Second example:
![image](https://user-images.githubusercontent.com/438911/123344158-17c25b00-d508-11eb-8eb5-91fd77489c4f.png)

**Comment by erictraut** [comment]
The signature help (when you start typing function arguments within parentheses) shows only two options.

The hover text (which is what is displayed in the screen shots above) displays all three declarations — the overloads and the implementation — if they are provided. Likewise, the "go to definition" command displays all three locations.

I think it's reasonable to hide the implementation's signature in the hover text for overloaded functions. TypeScript does this, so there is precedent.

I've updated the logic in the hover text provider accordingly. This will be included in the next release.

**Comment by glacials** [comment]
Thank you! That was a very fast fix and I appreciate it a lot ❤️

**Comment by jakebailey** [comment]
This issue has been fixed in version 2021.7.0, which we've just released. You can find the changelog here: https://github.com/microsoft/pylance-release/blob/main/CHANGELOG.md#202170-6-july-2021
